BubbleSort.java

```diff
@@ -0,0 +1,40 @@
+
+public class BubbleSort {
+
+    public void arrayDec(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            array[i] = i + 1;
+        }
+    }
+
+    public void arrayAsc(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            array[i] = array.length - i;
+        }
+    }
+
+    public void printArray(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            System.out.print(array[i] + " ");
+        }
+    }
+
+    public void bubbleSorting(int[] array) {
+        int n = array.length;
+        int compareNo = 0;
+        int temp = 0;
+
+        for (int i = 0; i < n - 1; i++) {
+            for (int j = 0; j < n - i - 1; j++) {
+                if (array[j] > array[j + 1]) {
+                    temp = array[j];
+                    array[j] = array[j + 1];
+                    array[j + 1] = temp;
+                    compareNo++;

```

Comment:
Dla posortowanych tablic wyświetla tobie 0 , ponieważ instrukcja w tej linijce powinna się znajdować przed ifem, a nie w jego środku ;)  Chcemy zliczać wszystkie porównania, niezależnie od tego czy w efekcie prównania zamieniamy 2 liczby czy nie. A teraz porównanie będzie zliczone tylko, gdy w jego efekcie dochodzi do zmiany liczb. Jeśli wyciąniesz tę instrukcję przed ifa, to zobaczysz, że dla niektórych tablic wychodzi więcej porównań Wynika to z tego, że jeszcze możesz wprowadzić optymalizację związaną z przerywaniem sortowania gdy nie doszło do zmian podczas jednego przejścia pętli (tej zewnętrznej pętli).